cassandane/Cassandane/Cyrus/JMAPMailbox.pm

```diff
@@ -4188,11 +4189,19 @@ EOF
     # T3 in (a,b) unseen
     # T4 in (a,trash) seen
 
-    $self->_check_counts('Initial Test',
-        a => [ 5, 2, 4, 2 ],
-        b => [ 3, 1, 2, 2 ],
-        trash => [ 1, 0, 1, 0 ],
-    );
+    if ($maj > 3 || ($maj == 3 && $min >= 7)) {
```

Comment:
This feels like it should be backported to 3.6 - since it's a bugfix for JMAP behaviour.